src/nvim/terminal.c

```diff
@@ -619,6 +619,24 @@ static void buf_set_term_title(buf_T *buf, char *title)
                &err);
 }
 
+// Get the associated buffer for the terminal and return whether it's in a
+// valid state to be used
+static bool term_buf_valid(Terminal *term, buf_T **buf)
+  FUNC_ATTR_NONNULL_ALL
+{
+  *buf = handle_get_buffer(term->buf_handle);
+  bool valid = true;
+  if (!*buf || !(valid = buf_valid(*buf))) {
+    // Destroyed by `close_buffer`. Do not do anything else.
+    if (!valid) {
+      term->buf_handle = 0;
```

Comment:
I mean `buf_T.terminal`. I'm thinking of `terminal_destroy()` which won't set `buf->terminal = NULL` if buf_handle is invalid. It seems like we're doing different things in different places, no single source of truth.

I haven't looked closely at all of this yet so I could be off base.